daffodil-cli/src/test/scala/org/apache/daffodil/CLI/Util.scala

```diff
@@ -228,11 +227,12 @@ object Util {
         (toIn, fromOut, fromErr, Left(thread))
       }
 
+    // Create an ExpectIt object that reads/writes the streams created above
     val eb = new ExpectBuilder()
     eb.withOutput(toIn)
     eb.withInputs(fromOut, fromErr)
     eb.withInputFilters(replaceInString("\r\n", "\n"))
-    eb.withTimeout(timeout, TimeUnit.SECONDS)
+    eb.withInfiniteTimeout()
```

Comment:
I'd prefer if we wrote our own thing that waits forever, but  that polls every 10ms or so, and after 2 seconds starts outputting messages to stderr or something. It can output 1 message every 5 seconds or something so as not to generate massive output if things do hang. 

The other wait forever loop and this one should identify themselves in the messages they display so we can tell them apart.

Reply:
That's kind of what happens now, but I think it's actually more useful than a "thread taking too long" kind of message.

With this change, we tell expect to never timeout (i.e. infinite timeout), but then run all the expect logic in a separate thread (the new TestThread class). Once the CLI is done, we give `timeout` seconds for the test thread to finish up all its checks and interrupt the thread if it reaches that timeout. That interrupt causes the test thread to die, we capture the exception caused by the interrupt, and rethrow it in the main thread for sbt to output.

So essentially we tell expect to not do timeout logic and we do it ourselves. The idea being that we don't want expect to timeout until after the CLI is done. Otherwise long CLI startup times can lead to expect failures.

Note that if it does hit that timeout (e.g. if I add a long `Thread.sleep` in a test body), we get something like this:

```
Test org.apache.daffodil.cliTest.TestCLIparsing.test_3677_CLI_Parsing_elementFormDefault_qualified failed: java.lang.InterruptedException: sleep interrupted, took 11.973 sec
[error]     at java.lang.Thread.sleep(Native Method)
[error]     at org.apache.daffodil.cliTest.TestCLIparsing.$anonfun$test_3677_CLI_Parsing_elementFormDefault_qualified$1(TestCLIParsing.scala:34)
```

Or if a test body expects something that doesn't exist, we get this:
```
[error] Test org.apache.daffodil.cliTest.TestCLIparsing.test_3677_CLI_Parsing_elementFormDefault_qualified failed: net.sf.expectit.ExpectIOException: Expect operation fails (timeout: -1 ms) for matcher: contains('<tns:e1>does not exist</tns:e1>'), took 4.863 sec
[error]     at net.sf.expectit.ExpectImpl.expectIn(ExpectImpl.java:106)
[error]     at net.sf.expectit.AbstractExpectImpl.expectIn(AbstractExpectImpl.java:57)
[error]     at net.sf.expectit.AbstractExpectImpl.expect(AbstractExpectImpl.java:61)
[error]     at org.apache.daffodil.CLI.Util$CLITester.expect(Util.scala:464)
[error]     at org.apache.daffodil.cliTest.TestCLIparsing.$anonfun$test_3677_CLI_Parsing_elementFormDefault_qualified$1(TestCLIParsing.scala:34)
```

In both of these examples, we get a line number that says exactly which line in the testBody expect was at when it was interrupted.

I'll add some comments to make it more clear and why this infinite timeout isn't really infinite.